internal/uvm/vsmb.go

```diff
@@ -203,7 +206,83 @@ func (uvm *UtilityVM) GetVSMBUvmPath(ctx context.Context, hostPath string, readO
 	if err != nil {
 		return "", err
 	}
-	return filepath.Join(share.guestPath, f), nil
+	return filepath.Join(share.GuestPath, f), nil
+}
+
+var _ = (Cloneable)(&VSMBShare{})
+
+// serializes the VSMBShare struct
+func (vsmb *VSMBShare) GobEncode() ([]byte, error) {
```

Comment:
We should probably encode/decode a version number here as well, and fail if the decoded version doesn't match the version in the code. That will allow us to make changes to the encoded schema.

Reply:
You mean maintain a separate version number for each cloneable resource?

Reply:
I have fixed this in the next PR #840

Reply:
Can we move the change into here? We should make sure this PR represents the set of work needed to support cloning UVMs. We don't really get any benefit from splitting them up if they are interdependent.